core/src/main/java/hudson/model/Run.java

```diff
@@ -659,20 +679,42 @@ public final long getStartTimeInMillis() {
         return startTime;
     }
 
+    //TODO(oleg_nenashev): Do we want to export full description by default taking the size considerations? Looks like it should be another REST API or a flag
     @Exported
+    @CheckForNull
     public String getDescription() {
         return description;
     }
 
+    /**
+     * Gets summary of the run.
+     * This is a formatted string, {@link hudson.markup.MarkupFormatter} should be applied to it in the WebUI.
+     * If the {@link #summary} field is {@code null}, the method will also try to retrieve a truncated {@link #description}.
+     * @since TODO
+     */
+    @Exported
+    @CheckForNull
+    public String getSummary() {
+        return summary != null ? summary : getTruncatedDescription();
```

Comment:
do you need to check if it's empty, or will it always be safely null here?

Reply:
I would prefer a strict data model here (good for getter performance), but I can add String truncation on the setter side